src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -776,6 +776,12 @@ public async Task VersionHandshakeAsync(NetworkPeerRequirement requirements, Can
 
                     await this.SendMessageAsync(addrPayload, cancellationToken).ConfigureAwait(false);
                 }
+
+                // We treat outbound peers as slightly more trusted, so we ask them for the peers they're aware of to aid our own discovery.
```

Comment:
As far as I remember processing will happen without checking if we've asked for addresses so if the peer is not trustworthy it can send us addresses on it's own. 
Therefore asking an outbound peer doesnt add more security. 

So I'd remove the inbound check.

Also dont we need a check that we have sufficient amount of addresses? If we already aware about 1000 peers for example do we really need to continue discovery?

Reply:
That's a fair point regarding sufficient known peers, although I think that having fresh options available is generally desirable. 999 of the 1000 could be old peers on dynamic IPs, for example. It doesn't cost much in network traffic to continue discovery.